src/server/game/Spells/Spell.cpp

```diff
@@ -7597,6 +7598,9 @@ WorldObjectSpellConeTargetCheck::WorldObjectSpellConeTargetCheck(float coneAngle
 
 bool WorldObjectSpellConeTargetCheck::operator()(WorldObject* target)
 {
+    if (_caster->GetTypeId() == TYPEID_PLAYER && target->IsInDist(_caster, _caster->GetBoundaryRadius()))
```

Comment:
If the target is within casters boundray radius, the cone effect will hit the target. (Even if the target is behind, but in boundary radius)